src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -16,7 +16,7 @@ namespace System.IO.Compression
     /// </summary>
     internal sealed class Deflater : IDisposable
     {
-        private ZLibNative.ZLibStreamHandle _zlibStream;
+        private ZLibNative.ZLibStreamHandle _zlibStream = null!; // Initialized by a method that is called from the constructor
```

Comment:
It seems like `DeflateInit` is only used in 1 place, you can inline and also make this readonly.

Reply:
I saw DeflateInit is used once, but it was not only initializing, plenty of code handling errors which i preffered to keep separate. But moved it as i don't have so strong opinion on keeping separate